vendor/bat-native-ledger/src/bat/ledger/internal/contribution/phase_one.cc

```diff
@@ -262,34 +270,33 @@ void PhaseOne::ReconcilePayloadCallback(
     if (response_status_code == net::HTTP_REQUESTED_RANGE_NOT_SATISFIABLE) {
       Complete(ledger::Result::CONTRIBUTION_AMOUNT_TOO_LOW,
                viewing_id,
-               reconcile.type_);
+               reconcile.type);
     } else {
       contribution_->AddRetry(ledger::ContributionRetry::STEP_PAYLOAD,
                               viewing_id);
     }
     return;
   }
 
-  braveledger_bat_helper::TRANSACTION_ST transaction;
-  bool success = braveledger_bat_helper::getJSONTransaction(response,
-                                                            &transaction);
+  ledger::TransactionProperties transaction;
+  const ledger::TransactionState transaction_state;
```

Comment:
will ping you about this one over slack

Reply:
as said on the call let's revert all this to bat_helpers

Reply:
Fixed and moved code to relevant state classes to use base::JSONReader